lib/evmone/instructions.hpp

```diff
@@ -890,6 +890,22 @@ inline code_iterator swapn(StackTop stack, ExecutionState& state, code_iterator
     return pos + 2;
 }
 
+inline evmc_status_code dataload(StackTop stack, ExecutionState& state) noexcept
+{
+    auto& index = stack.top();
+
+    if (state.data.size() < 32 || (state.data.size() - 32) < index)
+        return EVMC_INVALID_MEMORY_ACCESS;  // TODO better error
+
+    const auto begin = static_cast<size_t>(index);
+    uint8_t data[32] = {};
+    for (size_t i = 0; i < 32; ++i)
+        data[i] = state.data[begin + i];
+
+    index = intx::be::load<uint256>(data);
```

Comment:
Is this taking an array (and not iterators)? Otherwise could just use
```suggestion
    index = intx::be::load<uint256>(begin + index);
```

Reply:
It takes array

Reply:
Need an overload in intx 😅

Reply:
Made an issue: https://github.com/chfast/intx/issues/288

Reply:
Use `intx::be::unsafe::load(&state.data[begin])`.